testdata/project-v3-with-deploy-image/controllers/busybox_controller.go

```diff
@@ -112,8 +127,29 @@ func (r *BusyboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if isBusyboxMarkedToBeDeleted {
 		if controllerutil.ContainsFinalizer(busybox, busyboxFinalizer) {
 			log.Info("Performing Finalizer Operations for Busybox before delete CR")
+
+			// The following implementation will update the status
+			meta.SetStatusCondition(&busybox.Status.Conditions, metav1.Condition{Type: typeDegradedBusybox,
```

Comment:
/hold 

we need to fetch all the Memcached before to do the update 
therefore, the best approach is ensure that we have a method that does what we do here [https://github.com/kubernetes-sigs/kubebuilder/blob/149d3a2acd7a76ead0a54afd6ef4b5[…]/project-v3-with-deploy-image/controllers/busybox_controller.go](https://github.com/kubernetes-sigs/kubebuilder/blob/149d3a2acd7a76ead0a54afd6ef4b583ea505911/testdata/project-v3-with-deploy-image/controllers/busybox_controller.go#L85-L100)

and we will call it so we do not duplicate code

c/c @NikhilSharmaWe